### PR TITLE
Sftp file stream should consider negative offsets in Seek

### PR DESCRIPTION
**Pull request by S-Jindal**
#1018 

The sftp file stream must consider negative offsets in Seek method when the caller wants to read from the end of the file using `SeekOrigin.End`

### CODE REVIEW

File: src/Renci.SshNet/Sftp/SftpFileStream.cs [hunk 1]
```diff
@@ -6,6 +6,7 @@
 #if FEATURE_TAP
 using System.Threading.Tasks;
 #endif
+using System.Reflection;
```
Comment (zybexXL) [inline src/Renci.SshNet/Sftp/SftpFileStream.cs:9]: What's this for?

File: src/Renci.SshNet/Sftp/SftpFileStream.cs [hunk 2]
```diff
@@ -798,15 +801,16 @@ public override long Seek(long offset, SeekOrigin origin)
...
                         case SeekOrigin.End:
-                            var attributes = _session.RequestFStat(_handle, false);
-                            newPosn = attributes.Size - offset;
+                            attributes = _session.RequestFStat(_handle, false);
+                            newPosn = attributes.Size + offset;
                             break;
                     }
 
-                    if (newPosn == -1)
+                    if (newPosn == -1 || (attributes != null && newPosn > attributes.Size))
```
Comment (zybexXL) [inline src/Renci.SshNet/Sftp/SftpFileStream.cs:809]: position > size is valid, it's a way to extend the file size

**Comment by zybexXL** [comment]
PR #910 already fixes this same issue and also refactors the code in a better way. This PR is not needed.

File: src/Renci.SshNet/Sftp/SftpFileStream.cs (hunk 2 above)
Comment (zybexXL) [inline src/Renci.SshNet/Sftp/SftpFileStream.cs:809]: seeking past the end of file when writing is valid, it's a way to extend the file size.

File: src/Renci.SshNet/Sftp/SftpFileStream.cs (hunk 1 above)
Comment (S-Jindal) [reply src/Renci.SshNet/Sftp/SftpFileStream.cs:9]: Dummy change, will remove it

**Comment by S-Jindal** [comment]
> PR #910 already fixes this same issue and also refactors the code in a better way. This PR is not needed.

@zybexXL  why isn't that PR merged yet? It was opened 8 months ago. I will abandon this in case author is willing to merge that

**Comment by zybexXL** [comment]
> @zybexXL why isn't that PR merged yet? It was opened 8 months ago. I will abandon this in case author is willing to merge that

Good question, ask the maintainer @drieseng. There are dozens of pending PRs to merge, some for much longer than 910. This PR would have the same fate. I just merge whatever I need and build my own version instead of using the published NuGet packages.

Would someone please consider forking this to a more active project? It's clear the owner no longer has the time for this and is unwilling to add other maintainers. Think of the children!

**Comment by S-Jindal** [comment]
> > @zybexXL why isn't that PR merged yet? It was opened 8 months ago. I will abandon this in case author is willing to merge that
> 
> Good question, ask the maintainer @drieseng. There are dozens of pending PRs to merge, some for much longer than 910. This PR would have the same fate. I just merge whatever I need and build my own version instead of using the published NuGet packages.
> 
> Would someone please consider forking this to a more active project? It's clear the owner no longer has the time for this and is unwilling to add other maintainers. Think of the children!

Ok I will try to email Gert. If he responds I will ask him to add at least one active maintainer with "owner permissions". But if he doesn't respond, would you be willing to fork it and maintain actively? And are you familiar with the release process on nuget. In .NET it isn't possible to pull develop version of the project, we need to release on nuget

**Comment by zybexXL** [comment]
I actually started a fork a few months back, but then decided not to go ahead for fear of not having time to maintain it (as Gert doesn't) as I'm quite busy too nowadays. I would however have no problems adding several other maintainers to the project. I may reconsider, if this project stays in zombie mode.

**Comment by S-Jindal** [comment]
Gert has responded over email that is ready to add other maintainers to this project. @zybexXL  please share your email to be part of that conversation. 
Let's pull @IgorMilavec  as well. I can he has contributed to several areas in this project

**Comment by zybexXL** [comment]
Thanks @S-Jindal, I've sent an email to you and Gert.

**Comment by IgorMilavec** [comment]
@S-Jindal sure, I can help out. But unless we set project governance in a drastically more open way, I'm afraid it will just raise expectations and provide little value. 

I think our goal should be to provide value to SSH.NET users, specifically by:
1. Releasing often (every commit to main should trigger a preview NuGet package build)
2. Eliminating the gravest bugs (non-compliance with established interfaces)
3. Supporting current platform versions

If @drieseng agrees, we should have a call to discuss project governance.

**Comment by drieseng** [comment]
Duplicate of https://github.com/sshnet/SSH.NET/pull/910.

**Pull request closed without merging**